provider/haproxy/haproxy.go

```diff
@@ -278,16 +278,39 @@ func (lbp *Provider) IsEndpointDrained(ep *config.Endpoint) bool {
 	return false
 }
 
-func (dm *drainMgr) readCurrentStats(ep *config.Endpoint) (string, string, error) {
+func (dm *drainMgr) readCurrentStats(ep *config.Endpoint) (string, []string, error) {
+	totalScur := 0
+	var backendNames []string
 	//read stats
 	stats, err := dm.ReadStats()
 	if err != nil {
-		return "", "", fmt.Errorf("Error %v reading haproxy stats on  %v", err, dm.haproxySock)
+		return "", backendNames, fmt.Errorf("Error %v reading haproxy stats on  %v", err, dm.haproxySock)
 	}
-	if endpointStat, ok := stats[ep.Name]; ok {
-		return endpointStat["scur"], endpointStat["pxname"], nil
+
+	if _, ok := stats[ep.Name]; !ok {
+		return "", backendNames, fmt.Errorf("Error %v finding endpoint %v in haproxy stats", err, ep.Name)
+	}
+
+	for _, endpointStat := range stats[ep.Name] {
+		logrus.Debugf("scur for %v is: %v", endpointStat["pxname"]+"/"+ep.Name, endpointStat["scur"])
+		if escur, ok := endpointStat["scur"]; ok {
```

Comment:
should we handle !ok case here?

Reply:
no need to handle !ok case, since in that case, "totalScur" for that IP will remain unchanged, but we need to proceed to 306 to find each backend name for this IP - so that all backends can be set to weight 0.